src/Meshes/Builders/goldbergBuilder.ts

```diff
@@ -0,0 +1,606 @@
+            const center: Vector2 = uvRange[i][2]
+            const radius: number = uvRange[i][3];
+            const angle: number = uvRange[i][4];
+            const points5: number[] = [];
+            const points6: number[] = [];
+            let u:number;
+            let v: number;
+            for (let p = 0; p < 5; p++ ) {
+                u = center.x + radius * Math.cos(angle + p * Math.PI / 2.5);
+                v = center.y + radius * Math.sin(angle + p * Math.PI / 2.5);
+                if ( u < 0) {
+                    u = 0;
+                }
+                if (u > 1) {
+                    u = 1;
+                }
+                points5.push(u, v);
+            }
+            for (let p = 0; p < 6; p++ ) {
+                u = center.x + radius * Math.cos(angle + p * Math.PI / 3);
+                v = center.y + radius * Math.sin(angle + p * Math.PI / 3);
+                if ( u < 0) {
+                    u = 0;
+                }
+                if (u > 1) {
+                    u = 1;
+                }
+                points6.push(u, v);
+            }
+            for ( let f = min; f < Math.min(12, max + 1); f++ ) {
+                for (let p = 0; p < 5; p++) {
+                    uvs[10 * f + 2 * p] = points5[2 * p];
+                    uvs[10 * f + 2 * p + 1] = points5[2 * p + 1];
+                }
+            }
+            for ( let f = Math.max(12, min); f < max + 1; f++ ) {
+                for (let p = 0; p < 6; p++) {
+                    //120 + 12 * (f - 12) = 12 * f - 24
+                    uvs[12 * f - 24 + 2 * p] = points6[2 * p];
+                    uvs[12 * f - 23 + 2 * p] = points6[2 * p + 1];
+                }
+            }
+        }
+
+        return uvs;
+    }
+
+    public setFaceUVs(uvRange : any[][]) {
+        const newUVs: FloatArray = this._changeFaceUVs(uvRange);
+        this.setVerticesData(VertexBuffer.UVKind, newUVs);
+    };
+
+    public updateFaceUVs(uvRange : any[][]) {
+        const newUVs = this._changeFaceUVs(uvRange);
+        this.updateVerticesData(VertexBuffer.UVKind, newUVs);
+    }
+
+    public placeOnFaceAt(mesh: Mesh, face: number, position: Vector3) {
+		const orientation = Vector3.RotationFromAxis(this._faceXaxis[face], this._faceYaxis[face], this._faceZaxis[face]);
+		mesh.rotation = orientation;
+		mesh.position = this._faceCenters[face].add(this._faceXaxis[face].scale(position.x)).add(this._faceYaxis[face].scale(position.y)).add(this._faceZaxis[face].scale(position.z));
+	}
+};
+
+Mesh.CreateGoldberg = (name: string, options: { m?: number, n: number, size?: number, sizeX?: number, sizeY?: number, sizeZ?: number, updatable?: boolean, sideOrientation?: number }, scene: Scene): Goldberg => {
+    return GoldbergBuilder.CreateGoldberg(name, options, scene);
+}; */
+
+/**
+ * Class containing static functions to help procedurally build a Goldberg mesh
+ */
+ export class GoldbergBuilder {
+    /**
+     * Creates the Mesh for a Geodesic Polyhedron

```

Comment:
The comment sounds wrong -> should it not be about Goldberg also a quick link to what Goldberg is ? and I guess it should be renamed to GoldbergPolyhedron ?

Reply:
Done